src/extension.ts

```diff
@@ -2812,10 +2787,8 @@ function _show_skip_taskbar_windows(ext: Ext) {
  *
  */
 function _hide_skip_taskbar_windows() {
-    if (!GNOME_VERSION?.startsWith("4")) {
-        if (default_isoverviewwindow_ws)
-            Workspace.prototype._isOverviewWindow = default_isoverviewwindow_ws;
-    }
+    if (default_isoverviewwindow_ws)
+        Workspace.prototype._isOverviewWindow = default_isoverviewwindow_ws;
```

Comment:
I actually started reading the other code in this `_hide_skip_taskbar_windows()` function and it looks like the same bug exists in all the other un-hooking code below.

Which makes me wonder if perhaps the variables are being reset somewhere else.

Reply:
The first function show_skip_taskbar is the ENABLER, the second function is the DISABLER of what was introduced by the first one(it needs to be cleaned up when pop-shell gets removed or disabled by the user). I think what you are getting it is an indicator that the first one was called. However that's what the default_function variables are for.

Reply:
This is a _normal_ cleanup mechanism/workflow/pattern in gnome-extensions

Reply:
Yup, I think it is a good idea to cleanup the `default_<function>` vars when disable is called.

Reply:
@jmmaranan Yeah, what I mean is:

- The enabler is doing "if var X is missing, create var X and hook the Gnome function"
- The disabler is doing "if var X exists, unhook the gnome function" (it does not clean up var X)
- If you call the enabler again, var X still exists, so it doesn't re-hook.

If this is a correct reading of what it's doing, then it looks like all of the hook variables need to be cleaned up here.

Reply:
Yes, good catch!

Reply:
@Bananaman - the extension.ts is going to be scoped under the Extension GObject instance of it at runtime, and that gets desttroyed when disable by ExtensionSystem. But it doesn't hurt to clean them up.

Reply:
@jmmaranan Ahh, after seeing that nothing was cleaned up I began to suspect some kind of cleanup. But yeah, I agree it would be good to clean them up to avoid surprises. :)

Reply:
I added the explicit variable cleanup back to null